lib/segment/src/payload_storage/nested_query_checker.rs

```diff
@@ -136,12 +139,15 @@ pub fn check_nested_is_null_condition(
 }
 
 /// Return indexes of the elements matching the condition in the payload values
-pub fn nested_check_field_condition(
+pub fn nested_check_field_condition<R>(
     field_condition: &FieldCondition,
-    payload: &Payload,
+    payload: &impl PayloadContainer,
     nested_path: &JsonPathPayload,
-    field_indexes: &IndexesMap,
-) -> BitVec {
+    field_indexes: &HashMap<PayloadKeyType, R>,
+) -> BitVec
+where
+    R: AsRef<Vec<FieldIndex>>,
```

Comment:
I guess just using `&[FieldIndex]` would be better here, if possible.

We don't have a situation like with [`read_to_string`](https://doc.rust-lang.org/std/fs/fn.read_to_string.html) where we try to convert multiple types into `&FieldIndex`. At least, I cannot find it.